pkg/migrate/cstor/volume.go

```diff
@@ -0,0 +1,912 @@
+		csiPV.Spec.VolumeMode = pvObj.Spec.VolumeMode
+		return csiPV, true, nil
+	}
+	klog.Infof("PV %s already in csi form", pvObj.Name)
+	return pvObj, false, nil
+}
+
+func (v *VolumeMigrator) generateCSIPVFromCV(
+	cvName string,
+	pvcObj *corev1.PersistentVolumeClaim,
+) (*corev1.PersistentVolume, error) {
+	cvObj, err := cv.NewKubeclient().WithNamespace(v.CVNamespace).
+		Get(cvName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	csiPV := &corev1.PersistentVolume{}
+	csiPV.Name = cvObj.Name
+	csiPV.Spec.AccessModes = pvcObj.Spec.AccessModes
+	csiPV.Spec.ClaimRef = &corev1.ObjectReference{
+		APIVersion: pvcObj.APIVersion,
+		Kind:       pvcObj.Kind,
+		Name:       pvcObj.Name,
+		Namespace:  pvcObj.Namespace,
+	}
+	csiPV.Spec.Capacity = corev1.ResourceList{
+		corev1.ResourceStorage: cvObj.Spec.Capacity,
+	}
+	csiPV.Spec.PersistentVolumeSource = corev1.PersistentVolumeSource{
+		CSI: &corev1.CSIPersistentVolumeSource{
+			Driver:       cstorCSIDriver,
+			FSType:       cvObj.Annotations["openebs.io/fs-type"],
+			VolumeHandle: cvObj.Name,
+			VolumeAttributes: map[string]string{
+				"openebs.io/cas-type":                          "cstor",
+				"storage.kubernetes.io/csiProvisionerIdentity": csiProvisionerIdentity,
+			},
+		},
+	}
+	csiPV.Spec.PersistentVolumeReclaimPolicy = corev1.PersistentVolumeReclaimDelete
+	if v.StorageClass.ReclaimPolicy != nil {
+		csiPV.Spec.PersistentVolumeReclaimPolicy = *v.StorageClass.ReclaimPolicy
+	}
+	csiPV.Spec.StorageClassName = v.StorageClass.Name
+	csiPV.Spec.VolumeMode = pvcObj.Spec.VolumeMode
+	return csiPV, nil
+}
+
+func (v *VolumeMigrator) createCVC(pvName string) error {
+	var (
+		err    error
+		cvcObj *cstor.CStorVolumeConfig
+		cvObj  *apis.CStorVolume
+	)
+	cvcObj, err = v.OpenebsClientset.CstorV1().CStorVolumeConfigs(v.OpenebsNamespace).
+		Get(pvName, metav1.GetOptions{})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		return err
+	}
+	if k8serrors.IsNotFound(err) {
+		cvObj, err = cv.NewKubeclient().WithNamespace(v.CVNamespace).
+			Get(pvName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		annotations := map[string]string{
+			"openebs.io/volumeID":      pvName,
+			"openebs.io/volume-policy": pvName,
+		}
+		labels := map[string]string{
+			"openebs.io/cstor-pool-cluster": v.StorageClass.Parameters["cstorPoolCluster"],
+		}
+		if len(cvObj.Labels["openebs.io/source-volume"]) != 0 {
+			labels["openebs.io/source-volume"] = cvObj.Labels["openebs.io/source-volume"]
+		}
+		finalizer := "cvc.openebs.io/finalizer"
+		cvcObj = cstor.NewCStorVolumeConfig().
+			WithName(cvObj.Name).
+			WithNamespace(v.OpenebsNamespace).
+			WithAnnotations(annotations).
+			WithLabelsNew(labels).
+			WithFinalizer(finalizer)
+		cvcObj.Spec.Capacity = corev1.ResourceList{
+			corev1.ResourceName(corev1.ResourceStorage): cvObj.Spec.Capacity,
+		}
+		cvcObj.Spec.Provision.Capacity = corev1.ResourceList{
+			corev1.ResourceName(corev1.ResourceStorage): cvObj.Spec.Capacity,
```

Comment:
We need to fetch this value from CVR and need to raise a PR in cstor-operaters repo.
https://github.com/openebs/cstor-operators/blob/61da55f33bac8ccba78ef5781064287379bacfb9/pkg/controllers/cstorvolumeconfig/volume_operations.go#L479
Pass value from cvc.Spec.Provision.Capacity